service/gcs/runtime/runc/runc.go

```diff
@@ -376,7 +378,11 @@ func (r *runcRuntime) waitOnProcess(pid int) (oslayer.ProcessExitState, error) {
 }
 
 func (p *process) Wait() (oslayer.ProcessExitState, error) {
-	return p.c.r.waitOnProcess(p.pid)
+	state, err := p.c.r.waitOnProcess(p.pid)
+	if p.relay != nil {
```

Comment:
Since this extra functionality is being added directly to Wait() for *process, maybe Wait() for *container should call process.Wait rather than c.r.waitOnProcess.